### ISSUE TITLE
How can I disable Empty Responses from Handler?

### ISSUE THREAD

**Issue by HaidStefan** [plgd-dev/go-coap#188]
Hello,

I'm trying to send Non-Confirmable Messages from a Python Client to a Go-Server.
For sending the messages I'm using the aiocoap library in python. 

My server always stops with the following error if i start the client multiple times:
**2020/12/13 11:15:05 cannot read from udp connection: %!w(<nil>)**

<details>
<summary>Logfile - Click to expand</summary>
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000e3ff, Path: position from 192.168.0.27:51139
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000cc72, Path: position from 192.168.0.27:51140
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00003d5b, Path: position from 192.168.0.27:63282
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00000959, Path: position from 192.168.0.27:51141
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000b0a8, Path: position from 192.168.0.27:51142
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00000284, Path: position from 192.168.0.27:51143
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00001a07, Path: position from 192.168.0.27:51144
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000b0c5, Path: position from 192.168.0.27:51145
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000664e, Path: position from 192.168.0.27:63086
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000eda4, Path: position from 192.168.0.27:51146
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 000024ad, Path: position from 192.168.0.27:51147
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000badd, Path: position from 192.168.0.27:51148
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00004a24, Path: position from 192.168.0.27:51149
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000f35b, Path: position from 192.168.0.27:51150
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00005597, Path: position from 192.168.0.27:51151
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 0000fdc3, Path: position from 192.168.0.27:51152
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00003e7b, Path: position from 192.168.0.27:51153
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00003333, Path: position from 192.168.0.27:51154
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 00006c2b, Path: position from 192.168.0.27:51155
2020/12/13 11:32:31 Got message path=position: Code: POST, Token: 000040be, Path: position from 192.168.0.27:51156
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
keepalive: context deadline exceeded
2020/12/13 11:32:37 cannot read from udp connection: %!w(<nil>)

Process finished with exit code 1
</details>

I analyzed the network traffic and saw that the Server sends Empty, Confirmable Response Messages:
![](https://i.gyazo.com/9ccf6fc6b0b11b3faa5774705e17322f.png)

There are also a lot of "keepalive: context deadline exceeded" Loggings...
How can I disable sending Responses?
And is this even the Reason why the Server Stops with the **cannot read from udp connection: %!w(<nil>)** Error?

Currently, I'm using the following server code:

```golang
package main

import (
	coap "github.com/plgd-dev/go-coap/v2"
	"github.com/plgd-dev/go-coap/v2/message"
	"github.com/plgd-dev/go-coap/v2/mux"
	"log"
)

func getPath(opts message.Options) string {
	path, err := opts.Path()
	if err != nil {
		log.Printf("cannot get path: %v", err)
		return ""
	}
	return path
}

func main() {
	log.Fatal(coap.ListenAndServe("udp", ":5683",
		mux.HandlerFunc(func(w mux.ResponseWriter, r *mux.Message) {
			log.Printf("Got message path=%v: %+v from %v", getPath(r.Options), r, w.Client().RemoteAddr())
		})))
}
```

**Comment by jkralik** [comment]
You need to use disable keepalive feature via option udp.WithKeepAlive(nil). By default it is enabled.
Ofcourse you need to create server from udp package like https://github.com/plgd-dev/go-coap/blob/939f9dd26217635e1eb7ca5bbb237e5acbb61331/udp/server_test.go#L141.

**Comment by Niondir** [comment]
`cannot read from udp connection: %!w()` is mal formatted and is corrected in PR #190
In #191 the handling of NON messages was also improve and might help with your issue.

Let me know if you still face to problem with the lastest version when the above PRs are merged.

**Comment by Niondir** [comment]
The PR is merged. Please try again. The server should not answer when the request is non-confirmable
